src/utils/containsValidUrl.ts

```diff
@@ -0,0 +1,8 @@
+export default function containsValidUrl(content: string): boolean {
+  if (content.length === 0 || typeof content != 'string') return false;
+
+  // NOTE: there are likely more optimal ways to do this,
+  // see https://stackoverflow.com/questions/1500260/detect-urls-in-text-with-javascript
+  const validUrlRegex = /(\b(https?|ftp|file):\/\/[-A-Z0-9+&@#%?=~_|!:,.;]*[-A-Z0-9+&@#%=~_|])/gi;
```

Comment:
I don't think we need to pull a package in for this, but lets remove matching on `ftp` and `file` schemes, since those won't work in Discord.